src/core/containers/runtime/uv_provider.py

```diff
@@ -0,0 +1,183 @@
+"""Providers for launching Hugging Face Spaces via ``uv run``."""
+
+from __future__ import annotations
+
+import os
+import socket
+import subprocess
+import time
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+import requests
+
+from .providers import ContainerProvider
+
+
+def _poll_health(health_url: str, timeout_s: float) -> None:
+    """Poll a health endpoint until it returns HTTP 200 or times out."""
+
+    deadline = time.time() + timeout_s
+    while time.time() < deadline:
+        try:
+            response = requests.get(health_url, timeout=2.0)
+            if response.status_code == 200:
+                return
+        except requests.RequestException:
+            pass
```

Comment:
```suggestion
            timeout = max(0.0001, min(deadline - time.time(), 2.0))
            response = requests.get(health_url, timeout=timeout)
            if response.status_code == 200:
                return
        except requests.RequestException:
            continue
```

(nit) to avoid waiting 2s + 0.5s if `timeout_s` is less than 2s.